pkg/controller/oneagent/oneagent_controller.go

```diff
@@ -391,25 +394,29 @@ func newDaemonSetForCR(instance *dynatracev1alpha1.OneAgent) *appsv1.DaemonSet {
 	}
 }
 
-func newPodSpecForCR(instance *dynatracev1alpha1.OneAgent) corev1.PodSpec {
+func newPodSpecForCR(instance dynatracev1alpha1.BaseOneAgentDaemonSet) corev1.PodSpec {
 	trueVar := true
 
 	envVarImg := os.Getenv("RELATED_IMAGE_DYNATRACE_ONEAGENT")
 	img := "docker.io/dynatrace/oneagent:latest"
-	if instance.Spec.Image != "" {
-		img = instance.Spec.Image
+	if instance.GetOneAgentSpec().Image != "" {
+		img = instance.GetOneAgentSpec().Image
 	} else if envVarImg != "" {
 		img = envVarImg
 	}
 
 	sa := "dynatrace-oneagent"
-	if instance.Spec.ServiceAccountName != "" {
-		sa = instance.Spec.ServiceAccountName
+	if instance.GetOneAgentSpec().ServiceAccountName != "" {
+		sa = instance.GetOneAgentSpec().ServiceAccountName
 	}
 
-	args := instance.Spec.Args
-	if instance.Spec.Proxy != nil && (instance.Spec.Proxy.ValueFrom != "" || instance.Spec.Proxy.Value != "") {
-		args = append(instance.Spec.Args, "--set-proxy=$(https_proxy)")
+	args := instance.GetOneAgentSpec().Args
+	if instance.GetOneAgentSpec().Proxy != nil && (instance.GetOneAgentSpec().Proxy.ValueFrom != "" || instance.GetOneAgentSpec().Proxy.Value != "") {
+		args = append(instance.GetOneAgentSpec().Args, "--set-proxy=$(https_proxy)")
+	}
+
+	if _, ok := instance.(*dynatracev1alpha1.OneAgentIM); ok {
+		args = append(instance.GetOneAgentSpec().Args, "--set-infra-only=true")
```

Comment:
You don't want to override the proxy if set as well.

```suggestion
		args = append(args, "--set-infra-only=true")
```